src/fastfield/multivalued/writer.rs

```diff
@@ -312,6 +314,137 @@ impl<'a, C: Column> Iterator for MultivalueStartIndexIter<'a, C> {
     }
 }
 
+/// Writer for multi-valued (as in, more than one value per document)
+/// int fast field.
+///
+/// This `Writer` is only useful for advanced users.
+/// The normal way to get your multivalued int in your index
+/// is to
+/// - declare your field with fast set to `Cardinality::MultiValues`
+/// in your schema
+/// - add your document simply by calling `.add_document(...)`.
+///
+/// The `MultiValuedFastFieldWriter` can be acquired from the
+
+pub struct MultiValueU128FastFieldWriter {
+    field: Field,
+    vals: Vec<u128>,
+    doc_index: Vec<u64>,
+}
+
+impl MultiValueU128FastFieldWriter {
+    /// Creates a new `U128MultiValueFastFieldWriter`
+    pub(crate) fn new(field: Field) -> Self {
+        MultiValueU128FastFieldWriter {
+            field,
+            vals: Vec::new(),
+            doc_index: Vec::new(),
+        }
+    }
+
+    /// The memory used (inclusive childs)
+    pub fn mem_usage(&self) -> usize {
+        self.vals.capacity() * std::mem::size_of::<UnorderedTermId>()
+            + self.doc_index.capacity() * std::mem::size_of::<u64>()
```

Comment:
If we have this capacity * mem size item in many places of the code, it might be useful to have a helper funciton somewhere.
```
fn size_of_vec<T>(vec: &Vec<T>) -> usize
```
The point being that it would make sure more resilient to refactoring. Right now if sometimes change the content of the Vec, they might easily forget to update this funciton.

Reply:
We could add a trait, and implement it on Vec and our types
``` 
trait MemUsage{
    fn mem_usage(&self) -> usize;
}
```

Reply:
ok with that solution

Reply:
https://github.com/quickwit-oss/tantivy/issues/1573